Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -1612,17 +1637,29 @@ def enrich_offense_with_events(client: Client, offense: Dict, fetch_mode: str, e
         context_data[MIRRORED_OFFENSES_QUERIED_CTX_KEY][offense_id] = search_id
         safely_update_context_data(context_data, version, offense_ids=[offense_id])
     else:
-        events, failure_message = poll_offense_events_with_retry(client, search_id, int(offense_id))
-    events_fetched = sum(int(event.get('eventcount', 1)) for event in events)
+        events, failure_message = poll_offense_events_with_retry(client,
+                                                                 search_id,
+                                                                 offense,
+                                                                 expected_events,
+                                                                 fetch_mode,
+                                                                 events_columns,
+                                                                 events_limit,
+                                                                 )
+    events_fetched = get_events_count(events)
     print_debug_msg(f'Events fetched for offense {offense_id}: {events_fetched}/{events_count}.')
     offense['events_fetched'] = events_fetched
-    if events:
-        offense['events'] = events
-    else:
-        print_debug_msg(f'No events were fetched for offense {offense_id}'
+    
+    if not events or \
+            (fetch_mode == FetchMode.all_events.value and events_fetched < expected_events):
```

Comment:
Sometimes QRadar won't return all events.
Some events could be old, so you'll never get them.